### PR TITLE
Merge base-networking into master

### PR DESCRIPTION
**Pull request by thatging3rkid**
base-networking contains most of our low-level networking and packet handling code, it has been unit tested and is ready to be merged back into master

### CODE REVIEW

File: collections/hashtable.c [hunk 1]
```diff
@@ -0,0 +1,363 @@
+//
+// Created by Alex Kneipp on 1/25/20.
```
Comment (ahkneipp) [inline collections/hashtable.c:2]: I wrote this, therefore it must be perfect, right?

File: collections/hashtable.h [hunk 2]
```diff
@@ -0,0 +1,179 @@
+//
+// Created by Alex Kneipp on 1/25/20.
+//
+
+#ifndef INC_2020_CORE_CODE_HASHTABLE_H
+#define INC_2020_CORE_CODE_HASHTABLE_H
+
+#include "list.h"
+
+typedef struct HashTable * HashTable_t;
```
Comment (ahkneipp) [inline collections/hashtable.h]: This is a different idiom than the other Collections, should probably be changed, but that would require a large refactor

Comment (ahkneipp) [reply collections/hashtable.h]: Really up to you, this is my preferred Idiom, but consistency is also pretty important

File: network/constants.h [hunk 3]
```diff
@@ -1,116 +1,23 @@
...
  */
-#ifndef __CORE_NETWORKING_CONSTANTS
-#define __CORE_NETWORKING_CONSTANTS
-#include <stdint.h>
+#ifndef __CORE_NETWORK_CONSTANTS
+#define __CORE_NETWORK_CONSTANTS
 
-// Define the robot data type, this defines how the rest of the data in the
-// packet should be handled.
-typedef enum RobotDataType {
```
Comment (ahkneipp) [inline network/constants.h:14]: Where have these constants been moved to?

**Review by ahkneipp — CHANGES_REQUESTED** [review]
Honestly for a lot of this I just said "Yeah I trust that that was unit tested enough"

A few recommended changes but mostly I'm just trusting you and willing to deal with the bugs if/when they pop up later and asking for clarification

File: collections/hashtable.h (hunk 2 above)
Comment (thatging3rkid) [reply collections/hashtable.h]: fixed in bdce31e

File: network/constants.h (hunk 3 above)
Comment (thatging3rkid) [reply network/constants.h:14]: these types are outdated, they do not exist in network/constants.h anymore

**Review by ahkneipp — APPROVED** [review]
Looks good to me

**Pull request merged**